react/the_react_ecosystem/fetching-data.md

```diff
@@ -0,0 +1,297 @@
+
+In this lesson, we'll explore the ins and outs of fetching data in React, starting with the basics of making API calls, managing component state, and handling asynchronous operations using JavaScript's ~fetch~ function.
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- Understand how to make fetch requests in React components.
+- Using conditonals to render content depending on the state of the request (short circuiting).
+- Catching and handling errors.
+- Lifting requests up the component hierarchy.
+
+### A basic fetch request
+
+Before we dive into the specifics of fetching data in React, let's first revisit how we can use the fetch API to get data from a server.
+
+<!-- fetch is used to make requests to external APIs and returns a Promise that resolves to the response object representing the server's response. -->
+
+~~~js
+const image = document.querySelector("img");
+fetch("https://jsonplaceholder.typicode.com/photos", {
+  mode: "cors",
+})
+  .then((response) => response.json())
+  .then((response) => {
+    img.src = response[0].url;
+  })
+  .catch((error) => console.error(error));
+~~~
+
+We're making a request to the JSONPlaceholder API to retrieve an image. The response is then converted to JSON format using the json() method, and then we set the image src to the url embedded in that response object. The catch method is used to catch any errors that may occur during the request.
+
+### Using fetch in React components
+
+Now let's take a look at how we can incorporate fetch into a similar React component. One common use case is to fetch data from an API when a component mounts, so that the data can be displayed on screen.
+
+Whenever a component needs to make a request as it renders, it's often best to wrap that fetch inside of an effect.
+
+**https://codesandbox.io/s/top-simple-react-request-rd9psp?file=/src/styles.css**
+
+~~~jsx
+import { useEffect, useState } from "react";
+
+const Image = () => {
+  const [imageURL, setImageURL] = useState(null);
+
+  useEffect(() => {
+    fetch("https://jsonplaceholder.typicode.com/photos", { mode: "cors" })
+      .then((response) => response.json())
+      .then((response) => setImageURL(response[0].url))
+      .catch((error) => console.error(error));
+  }, []);
+
+  return (
+    imageURL && (
+      <>
+        <h1>An image</h1>
+        <img src={imageURL} alt={"placeholder text"} />
+      </>
+    )
+  );
+};
+
+export default Image;
+~~~
+
+useState lets us bundle data within a component and rerender the component whenever state is updated, and useEffect allows us to perform side effects on those components. Since we need to synchronize this component's state (the image URL) with an unpredictable external system like the fetch API, we need an effect. In this case we only want to fetch the URL once when the component is first mounted so we pass an empty array as a dependency to useEffect.
```

Comment:
Learners already know how `useState` and `useEffect` works. Wording suggestion: 
```suggestion
`useState` lets us add the `imageURL` state, whereas `useEffect` allows us to perform side effects. In this case, the side effect is fetching data from an external API. Since we only need to fetch once when component mounts, we use an empty dependency array.
```